src/screens/ToolStore.tsx

```diff
@@ -0,0 +1,323 @@
+import * as React from 'react';
+import {
+  Image,
+  Button,
+  View,
+  StyleSheet,
+  Text,
+  ActivityIndicator,
+  FlatList,
+} from 'react-native';
+import {useDispatch} from 'react-redux';
+import * as Sentry from '@sentry/react-native';
+import Toast from 'react-native-toast-message';
+import {AppDispatch} from '../reduxApp';
+import {GradientBtn} from './CartScreen';
+
+/**
+ * An example of how to add a Sentry Transaction to a React component manually.
+ * So you can control all spans that belong to that one transaction.
+ */
+const ToolStore = ({navigation}) => {
+  const dispatch = useDispatch();
+  const [toolData, setToolData] = React.useState<
+    | {
+        sku: string;
+        name: string;
+        image: string;
+        id: number;
+        type: string;
+        price: number;
+      }[]
+    | null
+  >(null);
+
+  const transaction = React.useRef(null);
+
+  React.useEffect(() => {
+    // Initialize the transaction for the screen.
+    transaction.current = Sentry.startTransaction({
+      name: 'Toolstore screen',
+      op: 'navigation',
+    });
+
```

Comment:
doesn't the span activity have to occur here?
```
span?.setData('json', json);
        span?.finish();
```

I feel like I'm reading that .useEffect finishes executing (and .finishes the transaction.current), and then afterwards loadData will get called and try to set a span on that transaction that finished already.

If I run it I could insert some logs but could you talk me through at all? or do a call.